### ISSUE TITLE
Requests to Elixir's package repository are failing

### ISSUE THREAD

**Issue by nickva** [linuxone-community-cloud/tickets#58]
In Apache CouchDB's CI worker s390x node we noticed requests to https://repo.hex.pm fail often. All the other architectures arm, x86_64, ppc64le work reliably.

Trying it by hand from the docker container I could reproduce it as:
```
wget https://repo.hex.pm/installs/hex-1.x.csv.signed
--2023-10-06 17:15:18--  https://repo.hex.pm/installs/hex-1.x.csv.signed
Resolving repo.hex.pm (repo.hex.pm)... 151.101.2.2, 151.101.66.2, 151.101.130.2, ...
Connecting to repo.hex.pm (repo.hex.pm)|151.101.2.2|:443... connected.
GnuTLS: The TLS connection was non-properly terminated.
Unable to establish SSL connection.
```

It fails when run directly on the node (not in docker). Also added `-d` for extra debugging.
```
linux1@linux1:~$ wget -d https://repo.hex.pm/installs/hex-1.x.csv.signed
DEBUG output created by Wget 1.20.3 on linux-gnu.

Reading HSTS entries from /home/linux1/.wget-hsts
URI encoding = ‘UTF-8’
Converted file name 'hex-1.x.csv.signed' (UTF-8) -> 'hex-1.x.csv.signed' (UTF-8)
--2023-10-06 18:34:45--  https://repo.hex.pm/installs/hex-1.x.csv.signed
Resolving repo.hex.pm (repo.hex.pm)... 2a04:4e42:600::514, 2a04:4e42:400::514, 2a04:4e42:200::514, ...
Caching repo.hex.pm => 2a04:4e42:600::514 2a04:4e42:400::514 2a04:4e42:200::514 2a04:4e42::514 151.101.194.2 151.101.130.2 151.101.66.2 151.101.2.2
Connecting to repo.hex.pm (repo.hex.pm)|2a04:4e42:600::514|:443... Closed fd 3
failed: Connection timed out.
```

The failure seems to be sporadic but happens often enough that every 2 or 3 CI rebuilds of the main branch fails there.

Even when it succeeds it seems to be quite a bit slower than on other architecture, about 3 seconds:

```
time wget https://repo.hex.pm/installs/hex-1.x.csv.signed
--2023-10-06 18:42:16--  https://repo.hex.pm/installs/hex-1.x.csv.signed
Resolving repo.hex.pm (repo.hex.pm)... 151.101.2.2, 151.101.66.2, 151.101.130.2, ...
Connecting to repo.hex.pm (repo.hex.pm)|151.101.2.2|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 350 [binary/octet-stream]
Saving to: 'hex-1.x.csv.signed.2'

hex-1.x.csv.signed.2                        100%[==========================================================================================>]     350  --.-KB/s    in 0s

2023-10-06 18:42:19 (66.7 MB/s) - 'hex-1.x.csv.signed.2' saved [350/350]


real	0m3.214s
user	0m0.015s
sys	0m0.004s
```

On PPC64LE worker with the same OS (Ubuntu 20.04) it's 50 msec only.
```
$ time wget https://repo.hex.pm/installs/hex-1.x.csv.signed
--2023-10-06 18:42:09--  https://repo.hex.pm/installs/hex-1.x.csv.signed
Resolving repo.hex.pm (repo.hex.pm)... 2a04:4e42:64::514, 199.232.146.2
Connecting to repo.hex.pm (repo.hex.pm)|2a04:4e42:64::514|:443... connected.
HTTP request sent, awaiting response... 200 OK
Length: 350 [binary/octet-stream]
Saving to: ‘hex-1.x.csv.signed.1’

hex-1.x.csv.signed.1                        100%[==========================================================================================>]     350  --.-KB/s    in 0s

2023-10-06 18:42:09 (14.9 MB/s) - ‘hex-1.x.csv.signed.1’ saved [350/350]


real	0m0.045s
user	0m0.000s
sys	0m0.010s
```

**Comment by bgilkes** [comment]
Referring this to our developer.  Response may not be until Monday as he is in different timezone.

**Comment by chee-ip** [comment]
It looks an SSL handshake failure issue between the client and the server. And also it seems the server's hostname access is behind CDN. If that's the case, do you think you can disable the CDN temporarily for debugging purpose?  

As for the longer time to access the file, I think it's related to the ISP and corporate firewall/proxy settings. There is nothing we can do.

**Comment by nickva** [comment]
Thanks for investigating @chee-ip!

Unfortunately this is an external package server for the Elixir and Erlang communities, not unlike https://pypi.org for Python. It's likely behind a CDN with DoS protection and such and we don't have access to modify or inspect it. It seems odd that other CI workers we have never seem to have an issue fetching packages from there only the Z Linux instance...

We had tried to mitigate it a bit on our side by maintain a Hex cache on each CI worker node, but it still doesn't seem to help. Sadly, I think we'll have to exclude the Z Linux (s390x) CI worker for now. Perhaps sometime in the future, in a year so, we can try to return it back to the pool and see if the issue persists...

Thanks again for taking a look! Closing the issue for now as there is not much that can be done.
